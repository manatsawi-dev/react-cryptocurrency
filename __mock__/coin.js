export const coin = {
  id: "bitcoin",
  symbol: "btc",
  name: "Bitcoin",
  image:
    "https://assets.coingecko.com/coins/images/1/large/bitcoin.png?1547033579",
  current_price: 38414,
  market_cap: 722614338899,
  market_cap_rank: 1,
  fully_diluted_valuation: 810897958013,
  total_volume: 94479295758,
  high_24h: 41757,
  low_24h: 34387,
  price_change_24h: -2043.15339795,
  price_change_percentage_24h: -5.05014,
  market_cap_change_24h: -32333557508.892456,
  market_cap_change_percentage_24h: -4.28289,
  circulating_supply: 18713700,
  total_supply: 21000000,
  max_supply: 21000000,
  ath: 64805,
  ath_change_percentage: -39.70923,
  ath_date: "2021-04-14T11:54:46.763Z",
  atl: 67.81,
  atl_change_percentage: 57519.58461,
  atl_date: "2013-07-06T00:00:00.000Z",
  roi: null,
  last_updated: "2021-05-22T11:15:12.892Z",
};

export const coins = [
  {
    id: "bitcoin",
    symbol: "btc",
    name: "Bitcoin",
    image:
      "https://assets.coingecko.com/coins/images/1/large/bitcoin.png?1547033579",
    current_price: 38414,
    market_cap: 722614338899,
    market_cap_rank: 1,
    fully_diluted_valuation: 810897958013,
    total_volume: 94479295758,
    high_24h: 41757,
    low_24h: 34387,
    price_change_24h: -2043.15339795,
    price_change_percentage_24h: -5.05014,
    market_cap_change_24h: -32333557508.892456,
    market_cap_change_percentage_24h: -4.28289,
    circulating_supply: 18713700,
    total_supply: 21000000,
    max_supply: 21000000,
    ath: 64805,
    ath_change_percentage: -39.70923,
    ath_date: "2021-04-14T11:54:46.763Z",
    atl: 67.81,
    atl_change_percentage: 57519.58461,
    atl_date: "2013-07-06T00:00:00.000Z",
    roi: null,
    last_updated: "2021-05-22T11:15:12.892Z",
  },
  {
    id: "ethereum",
    symbol: "eth",
    name: "Ethereum",
    image:
      "https://assets.coingecko.com/coins/images/279/large/ethereum.png?1595348880",
    current_price: 2447.77,
    market_cap: 286436489108,
    market_cap_rank: 2,
    fully_diluted_valuation: null,
    total_volume: 141930143365,
    high_24h: 2763.89,
    low_24h: 2154.87,
    price_change_24h: -237.026076,
    price_change_percentage_24h: -8.82846,
    market_cap_change_24h: -29237587702.046387,
    market_cap_change_percentage_24h: -9.26195,
    circulating_supply: 115977161.3115,
    total_supply: null,
    max_supply: null,
    ath: 4356.99,
    ath_change_percentage: -43.14902,
    ath_date: "2021-05-12T14:41:48.623Z",
    atl: 0.432979,
    atl_change_percentage: 571981.49266,
    atl_date: "2015-10-20T00:00:00.000Z",
    roi: {
      times: 84.36093857639716,
      currency: "btc",
      percentage: 8436.093857639717,
    },
    last_updated: "2021-05-22T11:17:32.839Z",
  },
];
