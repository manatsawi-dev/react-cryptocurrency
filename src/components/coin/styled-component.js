import styled from 'styled-components';

export const CoinWrapper = styled.div`
  max-width: 1000px;
  min-width: 430px;
`;

export const CoinContainer = styled.div`
  display: flex;
  padding: 1rem;
  align-items: center;
  justify-content: space-between;
`;

export const CoinImage = styled.img`
  height: 40px;
  width: 40px;
`;

export const CoinName = styled.p`
  width: 150px;
  padding: 1rem;
  @media (max-width: 920px) {
    display: none;
  }
`;

export const CoinHead = styled.div`
  display: flex;
  align-items: center;
`;

export const CoinSymbol = styled.p`
  width: 100px;
  @media (max-width: 920px) {
    padding: 1rem;
  }
`;

export const CoinCurrentPrice = styled.p`
  width: 110px;
  text-align: end;
`;

export const CoinVolume = styled.p`
  width: 150px;
  text-align: end;
  @media (max-width: 560px) {
    display: none;
  }
`;

export const CoinChange = styled.p`
  color: ${(props) => (props.increse ? 'green' : 'red')};
  width: 100px;
  text-align: end;
`;

export const CoinMarketCap = styled.p`
  width: 240px;
  text-align: end;
  @media (max-width: 790px) {
    display: none;
  }
`;
