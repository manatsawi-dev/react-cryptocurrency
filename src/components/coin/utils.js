export const getLocalStringText = (value) => {
  if (typeof value !== 'number') {
    return 0;
  }
  return value.toLocaleString();
};

export const getPriceChnageText = (priceChange) => {
  if (typeof priceChange !== 'number') {
    return 0;
  }
  return priceChange.toFixed(2);
};
