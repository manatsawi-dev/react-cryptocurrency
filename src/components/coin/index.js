import React from 'react';
import PropTypes from 'prop-types';
import {
  CoinWrapper,
  CoinContainer,
  CoinImage,
  CoinName,
  CoinSymbol,
  CoinCurrentPrice,
  CoinVolume,
  CoinChange,
  CoinMarketCap,
  CoinHead,
} from './styled-component';
import * as utils from './utils';

const componentName = (props) => {
  const {image, coinName, coinSymbol, price, volume, priceChange, marketCap} = props;

  return (
    <CoinWrapper id="coin-wrapper">
      <CoinContainer id="coin-row">
        <CoinHead id="coin-head">
          <CoinImage id="coin-image" src={image} alt="crypto" />
          <CoinName id="coin-name">{coinName}</CoinName>
        </CoinHead>
        <CoinSymbol id="coin-symbol">{coinSymbol}</CoinSymbol>
        <CoinCurrentPrice id="coin-current-price">${price}</CoinCurrentPrice>
        <CoinVolume id="coin-volume">${utils.getLocalStringText(volume)}</CoinVolume>
        <CoinChange id="coin-change" increase={priceChange > 0}>
          {utils.getPriceChnageText(priceChange)}%
        </CoinChange>
        <CoinMarketCap id="coin-market-cap">MKT Cap: ${utils.getLocalStringText(marketCap)}</CoinMarketCap>
      </CoinContainer>
    </CoinWrapper>
  );
};

componentName.propTypes = {
  image: PropTypes.string.isRequired,
  coinName: PropTypes.string.isRequired,
  coinSymbol: PropTypes.string.isRequired,
  price: PropTypes.number.isRequired,
  volume: PropTypes.number.isRequired,
  priceChange: PropTypes.number.isRequired,
  marketCap: PropTypes.number.isRequired,
};

export default componentName;
