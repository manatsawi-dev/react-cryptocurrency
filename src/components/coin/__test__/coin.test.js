/* eslint-disable no-console */
import React from 'react';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';
import Coin from '../index';
import {coin} from '../../../../__mock__/coin';

describe('<Coin/>', () => {
  const origConsole = console.error;
  beforeEach(() => {
    console.error = () => {};
  });
  afterEach(() => {
    console.error = origConsole;
  });

  it('render Coin snapshot correctly', () => {
    const wrapper = renderer.create(
      <Coin
        image={coin.image}
        coinName={coin.name}
        coinSymbol={coin.symbol}
        price={coin.current_price}
        volume={coin.total_volume}
        priceChange={coin.price_change_percentage_24h}
        marketCap={coin.market_cap}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renderer Coin check props correctly', () => {
    const wrapper = shallow(<Coin />);
    expect(wrapper.find('div').length).toEqual(0);
  });

  it('renderer Coin invalid props type correctly', () => {
    const wrapper = shallow(
      <Coin
        image={coin.image}
        coinName={coin.name}
        coinSymbol={coin.symbol}
        price="ABC"
        volume={coin.total_volume}
        priceChange={coin.price_change_percentage_24h}
        marketCap={coin.market_cap}
      />,
    );
    expect(wrapper.find('div').length).toEqual(0);
  });

  it('render Coin correctly', () => {
    const wrapper = shallow(
      <Coin
        image={coin.image}
        coinName={coin.name}
        coinSymbol={coin.symbol}
        price={coin.current_price}
        volume={coin.total_volume}
        priceChange={coin.price_change_percentage_24h}
        marketCap={coin.market_cap}
      />,
    );
    expect(wrapper.find('#coin-row').length).toEqual(1);
    expect(wrapper.find('#coin-image').length).toEqual(1);
    expect(wrapper.find('#coin-name').length).toEqual(1);
    expect(wrapper.find('#coin-symbol').length).toEqual(1);
    expect(wrapper.find('#coin-volume').length).toEqual(1);
    expect(wrapper.find('#coin-change').length).toEqual(1);
    expect(wrapper.find('#coin-market-cap').length).toEqual(1);

    wrapper.setProps({coinName: 'NewCoin'});
    expect(wrapper.find('#coin-name').text()).toEqual('NewCoin');
  });
});
