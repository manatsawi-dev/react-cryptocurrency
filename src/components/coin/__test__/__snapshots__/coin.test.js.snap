// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Coin/> render Coin snapshot correctly 1`] = `
<div
  className="sc-bdnxRM gOncIR"
  id="coin-wrapper"
>
  <div
    className="sc-gtsrHT bkruiP"
    id="coin-row"
  >
    <div
      className="sc-eCApnc huHUyo"
      id="coin-head"
    >
      <img
        alt="crypto"
        className="sc-dlnjwi llvFNJ"
        id="coin-image"
        src="https://assets.coingecko.com/coins/images/1/large/bitcoin.png?1547033579"
      />
      <p
        className="sc-hKFxyN dIqaQi"
        id="coin-name"
      >
        Bitcoin
      </p>
    </div>
    <p
      className="sc-jSFjdj jPeUKV"
      id="coin-symbol"
    >
      btc
    </p>
    <p
      className="sc-gKAaRy fFcWFZ"
      id="coin-current-price"
    >
      $
      38414
    </p>
    <p
      className="sc-iCoGMd kdtAGX"
      id="coin-volume"
    >
      $
      94,479,295,758
    </p>
    <p
      className="sc-fujyAs csHDML"
      id="coin-change"
    >
      -5.05
      %
    </p>
    <p
      className="sc-pNWdM lhxuWv"
      id="coin-market-cap"
    >
      MKT Cap: $
      722,614,338,899
    </p>
  </div>
</div>
`;
